### ISSUE TITLE
Improving Usability of Asset/Model Management with Grid Display and Changes Highlighting

### ISSUE THREAD

**Issue by ashledombos** [makehumancommunity/mpfb2#107]
**Is your feature request related to a problem? Please describe.**

While I don't consider it a problem, I find it challenging to navigate through large asset libraries, such as 'clothes', and the modeling elements of the 'model' category. Currently, all items are displayed in a single column, requiring significant scrolling and making it difficult to locate modified models or loaded assets, even though subcategories are masked by default and there is a search/filter form. 
The challenge of navigating becomes even more pronounced with larger libraries like 'clothes' or some subcategories of the 'model' category, where it can also sometimes be a lengthy process to scroll through and locate the modified body elements.

**Describe the solution you'd like**

I believe the user experience would benefit greatly from a change in the display mode. I propose that the interface displays a single thumbnail for each sub-category of 'assets' and 'model' categories. For example, a single thumbnail for the 'clothes' subcategory or the 'hands' subcategory (empty or with some default content); When clicking on it , it would display a grid of choices. Once a choice is made, the thumbnail would either display the chosen asset with the 'load/unequip' button (for clothes) or the modeling element with its corresponding adjustable slider(s) (for 'model'). This would streamline the navigation process, minimize the need for scrolling and allow for a quick and clear overview of the active and modified assets/models, which could be highlighted during grid mode display.

**Describe alternatives you've considered**

As an alternative, only the visibility of active and modified assets in the current column display could be improved by changing their color or adding a border, and we could consider the search function is sufficient avoiding too much scrolling. However, I believe that the grid display would be a really effective and intuitive solution.

**Additional context**

MPFB is an excellent plugin – in my opinion, the best available compared to other open source or paid options. I believe that implementing this feature would enhance its usability even further. I have created some illustrative images to provide a clearer picture of my feature request.

Current UX
![image](https://github.com/makehumancommunity/mpfb2/assets/513335/cdb1f9bc-1d36-4fc2-82f2-ca3a9ecf2663)
![image](https://github.com/makehumancommunity/mpfb2/assets/513335/eb2c8a67-f808-4c78-8d36-18f1306b2f4c)
A single column, loaded assets and modified body elements are not highlighted.

![image](https://github.com/makehumancommunity/mpfb2/assets/513335/52ac0cf3-883e-450e-8a55-2c354abba7f3)
A capture from another plugin, we could imagine the highlighted one would be a loaded asset or a modified model.

**Comment by joepal1976** [comment]
Sounds like a good suggestion. I'll have to research how to go about it technically though.

**Comment by joepal1976** [comment]
Also, thank you for providing such well-written and clear description in the request. This is very helpful.

**Comment by joepal1976** [comment]
Actually, that wasn't as difficult as I had expected. 

![grid2](https://github.com/makehumancommunity/mpfb2/assets/10105480/3cb0a114-716b-4e2b-82a9-c224b3d7a44a)

I'll clean this up a bit and add the same functionality to the model panels.

**Comment by joepal1976** [comment]
This will appear in tomorrow's nightly build, or can be fetched from git now.

**Comment by ashledombos** [comment]
I was on vacation and did not expect this to be taken in account so quickly, I’ll test it asap, thank you !

**Comment by ashledombos** [comment]
Hello @joepal1976 

I’m not sure if I did something wrong, but I don’t see any difference.
I downloaded zip file from github, extracted, and zipped again the mpfb directory, removed previous mpfb addon, and installed the new zip.
But when I increase the N window, it just stretches the assets boxes…

**Comment by joepal1976** [comment]
@ashledombos : That's very strange. I tested on two different computers now, one linux and one windows, both running blender 3.4.1.

Can you check in your local version of mpfb/ui/assetlibrary/assetlibrarypanel.py around line 40 and see if those lines look like these?

```
    def _draw_section(self, scene, layout):
        _LOG.enter()
        tot_width = bpy.context.region.width
        cols = max(1, math.floor(tot_width / 256))
        _LOG.debug("Number of UI columns to use", cols)
```

**Comment by ashledombos** [comment]
It's working now \o/
For some reason, one version of the plugin was installed in the global directory while the new installation was in the user directory. What's surprising is that I thought the user directory took precedence over the global one. Regardless, deleting the data from the global directory allowed the update to be recognized.

Thanks for the improvements, this will help a lot!

As for the visibility of active items (which I did not clearly talk about), do you think it could be possible to change the color of the button (for instance “unequip”) or the thickness of the text to emphasize them in some way?

![image](https://github.com/makehumancommunity/mpfb2/assets/513335/c9b22285-3a39-4dc9-a23e-09fcf3027225)

**Comment by joepal1976** [comment]
@ashledombos : Best I can do is set "alert" status on the operator:

![image](https://github.com/makehumancommunity/mpfb2/assets/10105480/73fc546d-e9d6-4450-bc04-8ce76b25c146)

The color, font style etc is hardcoded in blender and can't be changed on an operator.

But I pushed an update in https://github.com/makehumancommunity/mpfb2/commit/66ebb22a3e599fd619bbf2d37106ac63157a289a for setting the altert status. It's in main now if you want to test.

**Comment by ashledombos** [comment]
This is absolutely perfect now! Truly awesome work! If time permits and it's feasible, it would be fantastic to see alert statuses on model item sliders when their value differs from the default/zero. Thank you so much for all these improvements!

![image](https://github.com/makehumancommunity/mpfb2/assets/513335/e582c7b7-a4ce-43f2-90d8-5d371651f890)

**Comment by ashledombos** [comment]
I think it’s fixed with [3d36aa3](https://github.com/makehumancommunity/mpfb2/commit/3d36aa3bd6f530bd544349224e41f10ba4a45956)
